src/psyclone/psyir/nodes/codeblock.py

```diff
@@ -174,8 +192,45 @@ def get_symbol_names(self):
                                  Fortran2003.End_If_Stmt)):
                 # We don't want labels associated with loop or branch control.
                 result.append(node.string)
-
+        # Precision on literals requires special attention since they are just
+        # stored in the tree as str (fparser/#456).
+        for node in walk(parse_tree, (Fortran2003.Int_Literal_Constant,
+                                      Fortran2003.Real_Literal_Constant,
+                                      Fortran2003.Logical_Literal_Constant,
+                                      Fortran2003.Char_Literal_Constant)):
+            if node.items[1]:
+                result.append(node.items[1])
+        # Complex literals require even more special attention.
+        for node in walk(parse_tree, Fortran2003.Complex_Literal_Constant):
+            # A complex literal constant has a real part and an imaginary part.
+            # Each of these can have a kind.
+            for part in node.items:
+                if part.items[1]:
+                    result.append(part.items[1])
         return result
 
+    def reference_accesses(self, var_accesses: VariablesAccessInfo):
+        '''
+        Get all variable access information. Since this is a CodeBlock we
+        only know the names of symbols accessed within it but not how they
+        are accessed. Therefore we err on the side of caution and mark
+        them all as READWRITE, unfortunately, this will include the names of
+        any routines that are called.
+
+        TODO #2863 - it would be better to use AccessType.UNKNOWN here but
+        currently VariablesAccessInfo does not consider that type of access.
+
+        This method makes use of
+        :py:meth:`~psyclone.psyir.nodes.CodeBlock.get_symbol_names` and is
```

Comment:
Is the "~" intended?

Reply:
Yep, it tells Sphinx just to render `get_symbol_names` rather than include the full module information.